rl4co/models/nn/env_embeddings/context.py

```diff
@@ -146,6 +147,50 @@ def _state_embedding(self, embeddings, td):
         state_embedding = td["vehicle_capacity"] - td["used_capacity"]
         return state_embedding
 
+class VRPBContext(EnvContext):
+    """Context embedding for the Capacitated Vehicle Routing Problem (CVRP).
+    Project the following to the embedding space:
+        - current node embedding
+        - remaining capacity (vehicle_capacity - used_capacity)
+    """
+
+    def __init__(self, embed_dim):
+        super(VRPContext, self).__init__(
+            embed_dim=embed_dim, step_context_dim=embed_dim + 1
+        )
+
+    def _state_embedding(self, embeddings, td):
+        mask = (td["used_capacity_backhaul"] == 0)
+        used_capacity = torch.where(mask, td["used_capacity_linehaul"], td["used_capacity_backhaul"])
+        state_embedding = td["vehicle_capacity"] - used_capacity
+        return state_embedding
+    
+class MTVRPContext(VRPBContext):
+    """Context embedding for the Capacitated Vehicle Routing Problem (CVRP).
+    Project the following to the embedding space:
+        - current node embedding
+        - remaining capacity (vehicle_capacity - used_capacity)
+        - current time
+        - current route length
+        - if route should be open
+    """
+
+    def __init__(self, embed_dim):
+        super(VRPBContext, self).__init__(
+            embed_dim=embed_dim, step_context_dim=embed_dim + 4
+        )
+
+    def _state_embedding(self, embeddings, td):
+
+        capacity = super()._state_embedding(embeddings, td)
+        current_time = td["current_time"]
+        current_length = td["current_route_length"]
```

Comment:
How does the model understand whether there is a limit? 
In case there is no limit (say CVRP), then it will be the same as having VRPL, since the model does not know whether the constraint will be enforced or not

Reply:
You are right. The route length in the state_emdding should be the rest length, i.e., length limit-current length, instead of current length... It seems to be a mistake...